The job **`dist-apple-various`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/14038961370/job/39304073245) [(plain)](https://github.com/rust-lang-ci/rust/commit/2316b76d008a5455753b70e83d7c374dfde9b2fd/checks/39304073245/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Updating files:  98% (51607/52660)
Updating files:  99% (52134/52660)
Updating files: 100% (52660/52660)
Updating files: 100% (52660/52660), done.
Switched to a new branch 'try'
branch 'try' set up to track 'origin/try'.
##[endgroup]
[command]/usr/local/bin/git log -1 --format=%H
2316b76d008a5455753b70e83d7c374dfde9b2fd
##[group]Run src/ci/scripts/setup-environment.sh
src/ci/scripts/setup-environment.sh
---
file:.git/config remote.origin.url=https://github.com/rust-lang-ci/rust
file:.git/config remote.origin.fetch=+refs/heads/*:refs/remotes/origin/*
file:.git/config gc.auto=0
file:.git/config http.https://github.com/.extraheader=AUTHORIZATION: basic ***
file:.git/config branch.try.remote=origin
file:.git/config branch.try.merge=refs/heads/try
file:.git/config remote.upstream.url=https://github.com/rust-lang/rust
file:.git/config remote.upstream.fetch=+refs/heads/*:refs/remotes/upstream/*
file:.git/config submodule.library/backtrace.active=true
file:.git/config submodule.library/backtrace.url=https://github.com/rust-lang/backtrace-rs.git
file:.git/config submodule.library/stdarch.active=true
---
CMAKE_x86_64-apple-darwin = None
CMAKE_x86_64_apple_darwin = None
HOST_CMAKE = None
CMAKE = None
running: cd "/Users/runner/work/rust/rust/build/x86_64-apple-darwin/llvm/build" && CMAKE_PREFIX_PATH="" DESTDIR="" LC_ALL="C" "cmake" "/Users/runner/work/rust/rust/src/llvm-project/llvm" "-DCMAKE_OSX_ARCHITECTURES=x86_64" "-G" "Ninja" "-DLLVM_ENABLE_ASSERTIONS=OFF" "-DLLVM_UNREACHABLE_OPTIMIZE=OFF" "-DLLVM_ENABLE_PLUGINS=OFF" "-DLLVM_TARGETS_TO_BUILD=AArch64;AMDGPU;ARM;BPF;Hexagon;LoongArch;MSP430;Mips;NVPTX;PowerPC;RISCV;Sparc;SystemZ;WebAssembly;X86" "-DLLVM_EXPERIMENTAL_TARGETS_TO_BUILD=AVR;M68k;CSKY;Xtensa" "-DLLVM_INCLUDE_EXAMPLES=OFF" "-DLLVM_INCLUDE_DOCS=OFF" "-DLLVM_INCLUDE_BENCHMARKS=OFF" "-DLLVM_INCLUDE_TESTS=OFF" "-DLLVM_ENABLE_TERMINFO=OFF" "-DLLVM_ENABLE_LIBEDIT=OFF" "-DLLVM_ENABLE_BINDINGS=OFF" "-DLLVM_ENABLE_Z3_SOLVER=OFF" "-DLLVM_PARALLEL_COMPILE_JOBS=4" "-DLLVM_TARGET_ARCH=x86_64" "-DLLVM_DEFAULT_TARGET_TRIPLE=x86_64-apple-darwin" "-DLLVM_ENABLE_WARNINGS=OFF" "-DLLVM_INSTALL_UTILS=ON" "-DLLVM_ENABLE_ZLIB=ON" "-DLLVM_ENABLE_LIBXML2=OFF" "-DLLVM_VERSION_SUFFIX=-rust-1.87.0-nightly" "-DCMAKE_INSTALL_MESSAGE=LAZY" "-DCMAKE_C_COMPILER_LAUNCHER=sccache" "-DCMAKE_CXX_COMPILER_LAUNCHER=sccache" "-DCMAKE_C_COMPILER=/Users/runner/work/rust/rust/clang+llvm-15.0.7-x86_64-apple-darwin21.0/bin/clang" "-DCMAKE_CXX_COMPILER=/Users/runner/work/rust/rust/clang+llvm-15.0.7-x86_64-apple-darwin21.0/bin/clang++" "-DCMAKE_ASM_COMPILER=/Users/runner/work/rust/rust/clang+llvm-15.0.7-x86_64-apple-darwin21.0/bin/clang" "-DCMAKE_C_FLAGS=-ffunction-sections -fdata-sections -fPIC -m64 --target=x86_64-apple-macosx -mmacosx-version-min=10.12 -fdebug-prefix-map=/Users/runner/work/rust/rust=/rustc/llvm" "-DCMAKE_CXX_FLAGS=-ffunction-sections -fdata-sections -fPIC -m64 --target=x86_64-apple-macosx -mmacosx-version-min=10.12 -stdlib=libc++ -fdebug-prefix-map=/Users/runner/work/rust/rust=/rustc/llvm" "-DCMAKE_SHARED_LINKER_FLAGS= -static-libstdc++" "-DCMAKE_MODULE_LINKER_FLAGS= -static-libstdc++" "-DCMAKE_EXE_LINKER_FLAGS= -static-libstdc++" "-DLLVM_ENABLE_ZSTD=OFF" "-DCMAKE_INSTALL_PREFIX=/Users/runner/work/rust/rust/build/x86_64-apple-darwin/llvm" "-DCMAKE_ASM_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64 --target=x86_64-apple-macosx -mmacosx-version-min=10.12" "-DCMAKE_BUILD_TYPE=Release"
CMake Deprecation Warning at /Users/runner/work/rust/rust/src/llvm-project/cmake/Modules/CMakePolicy.cmake:6 (cmake_policy):
  The OLD behavior for policy CMP0116 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
---
CMAKE_x86_64-apple-darwin = None
CMAKE_x86_64_apple_darwin = None
HOST_CMAKE = None
CMAKE = None
running: cd "/Users/runner/work/rust/rust/build/x86_64-apple-darwin/lld/build" && CMAKE_PREFIX_PATH="" DESTDIR="" LC_ALL="C" "cmake" "/Users/runner/work/rust/rust/src/llvm-project/lld" "-DCMAKE_OSX_ARCHITECTURES=x86_64" "-G" "Ninja" "-DCMAKE_INSTALL_MESSAGE=LAZY" "-DCMAKE_C_COMPILER_LAUNCHER=sccache" "-DCMAKE_CXX_COMPILER_LAUNCHER=sccache" "-DCMAKE_C_COMPILER=/Users/runner/work/rust/rust/clang+llvm-15.0.7-x86_64-apple-darwin21.0/bin/clang" "-DCMAKE_CXX_COMPILER=/Users/runner/work/rust/rust/clang+llvm-15.0.7-x86_64-apple-darwin21.0/bin/clang++" "-DCMAKE_ASM_COMPILER=/Users/runner/work/rust/rust/clang+llvm-15.0.7-x86_64-apple-darwin21.0/bin/clang" "-DCMAKE_C_FLAGS=-ffunction-sections -fdata-sections -fPIC -m64 --target=x86_64-apple-macosx -mmacosx-version-min=10.12 -fdebug-prefix-map=/Users/runner/work/rust/rust=/rustc/llvm" "-DCMAKE_CXX_FLAGS=-ffunction-sections -fdata-sections -fPIC -m64 --target=x86_64-apple-macosx -mmacosx-version-min=10.12 -stdlib=libc++ -fdebug-prefix-map=/Users/runner/work/rust/rust=/rustc/llvm" "-DCMAKE_SHARED_LINKER_FLAGS= -static-libstdc++" "-DCMAKE_MODULE_LINKER_FLAGS= -static-libstdc++" "-DCMAKE_EXE_LINKER_FLAGS= -static-libstdc++" "-DLLVM_ENABLE_ZSTD=OFF" "-DLLVM_CMAKE_DIR=/Users/runner/work/rust/rust/build/x86_64-apple-darwin/llvm/lib/cmake/llvm" "-DLLVM_INCLUDE_TESTS=OFF" "-DCMAKE_INSTALL_PREFIX=/Users/runner/work/rust/rust/build/x86_64-apple-darwin/lld" "-DCMAKE_ASM_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64 --target=x86_64-apple-macosx -mmacosx-version-min=10.12" "-DCMAKE_BUILD_TYPE=Release"
CMake Deprecation Warning at /Users/runner/work/rust/rust/src/llvm-project/cmake/Modules/CMakePolicy.cmake:6 (cmake_policy):
  The OLD behavior for policy CMP0116 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
---
CMAKE_x86_64-apple-darwin = None
CMAKE_x86_64_apple_darwin = None
HOST_CMAKE = None
CMAKE = None
running: cd "/Users/runner/work/rust/rust/build/x86_64-apple-darwin/native/sanitizers/build" && CMAKE_PREFIX_PATH="" DESTDIR="" LC_ALL="C" "cmake" "/Users/runner/work/rust/rust/src/llvm-project/compiler-rt" "-DCMAKE_OSX_ARCHITECTURES=x86_64" "-G" "Ninja" "-DCMAKE_C_COMPILER_TARGET=x86_64-apple-darwin" "-DCOMPILER_RT_BUILD_BUILTINS=OFF" "-DCOMPILER_RT_BUILD_CRT=OFF" "-DCOMPILER_RT_BUILD_LIBFUZZER=OFF" "-DCOMPILER_RT_BUILD_PROFILE=OFF" "-DCOMPILER_RT_BUILD_SANITIZERS=ON" "-DCOMPILER_RT_BUILD_XRAY=OFF" "-DCOMPILER_RT_DEFAULT_TARGET_ONLY=ON" "-DCOMPILER_RT_USE_LIBCXX=OFF" "-DLLVM_CONFIG_PATH=/Users/runner/work/rust/rust/build/x86_64-apple-darwin/llvm/bin/llvm-config" "-DCMAKE_INSTALL_MESSAGE=LAZY" "-DCMAKE_C_COMPILER=/Users/runner/work/rust/rust/clang+llvm-15.0.7-x86_64-apple-darwin21.0/bin/clang" "-DCMAKE_CXX_COMPILER=/Users/runner/work/rust/rust/clang+llvm-15.0.7-x86_64-apple-darwin21.0/bin/clang++" "-DCMAKE_ASM_COMPILER=/Users/runner/work/rust/rust/clang+llvm-15.0.7-x86_64-apple-darwin21.0/bin/clang" "-DCMAKE_C_FLAGS=-ffunction-sections -fdata-sections -fPIC -m64 --target=x86_64-apple-macosx -fdebug-prefix-map=/Users/runner/work/rust/rust=/rustc/llvm" "-DCMAKE_CXX_FLAGS=-ffunction-sections -fdata-sections -fPIC -m64 --target=x86_64-apple-macosx -stdlib=libc++ -fdebug-prefix-map=/Users/runner/work/rust/rust=/rustc/llvm" "-DCMAKE_SHARED_LINKER_FLAGS= -static-libstdc++" "-DCMAKE_MODULE_LINKER_FLAGS= -static-libstdc++" "-DCMAKE_EXE_LINKER_FLAGS= -static-libstdc++" "-DCMAKE_INSTALL_PREFIX=/Users/runner/work/rust/rust/build/x86_64-apple-darwin/native/sanitizers" "-DCMAKE_ASM_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64 --target=x86_64-apple-macosx -mmacosx-version-min=10.12" "-DCMAKE_BUILD_TYPE=Release"
CMake Deprecation Warning at /Users/runner/work/rust/rust/src/llvm-project/cmake/Modules/CMakePolicy.cmake:6 (cmake_policy):
  The OLD behavior for policy CMP0116 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
---
ld: warning: '/Users/runner/work/rust/rust/build/x86_64-apple-darwin/native/sanitizers/build/lib/ubsan/CMakeFiles/RTUbsan.osx.dir/ubsan_type_hash_itanium.cpp.o' has malformed LC_DYSYMTAB, expected 13 undefined symbols to start at index 6, found 10 undefined symbols starting at index 6
ld: warning: '/Users/runner/work/rust/rust/build/x86_64-apple-darwin/native/sanitizers/build/lib/ubsan/CMakeFiles/RTUbsan.osx.dir/ubsan_type_hash.cpp.o' has malformed LC_DYSYMTAB, expected 7 undefined symbols to start at index 2, found 4 undefined symbols starting at index 2
ld: warning: '/Users/runner/work/rust/rust/build/x86_64-apple-darwin/native/sanitizers/build/lib/ubsan/CMakeFiles/RTUbsan.osx.dir/ubsan_handlers.cpp.o' has malformed LC_DYSYMTAB, expected 25 undefined symbols to start at index 73, found 22 undefined symbols starting at index 73
cargo:root=/Users/runner/work/rust/rust/build/x86_64-apple-darwin/native/sanitizers
running: cd "/Users/runner/work/rust/rust/build/x86_64-apple-darwin/native/sanitizers/build" && CMAKE_PREFIX_PATH="" DESTDIR="" LC_ALL="C" "cmake" "/Users/runner/work/rust/rust/src/llvm-project/compiler-rt" "-DCMAKE_OSX_ARCHITECTURES=x86_64" "-G" "Ninja" "-DCMAKE_C_COMPILER_TARGET=x86_64-apple-darwin" "-DCOMPILER_RT_BUILD_BUILTINS=OFF" "-DCOMPILER_RT_BUILD_CRT=OFF" "-DCOMPILER_RT_BUILD_LIBFUZZER=OFF" "-DCOMPILER_RT_BUILD_PROFILE=OFF" "-DCOMPILER_RT_BUILD_SANITIZERS=ON" "-DCOMPILER_RT_BUILD_XRAY=OFF" "-DCOMPILER_RT_DEFAULT_TARGET_ONLY=ON" "-DCOMPILER_RT_USE_LIBCXX=OFF" "-DLLVM_CONFIG_PATH=/Users/runner/work/rust/rust/build/x86_64-apple-darwin/llvm/bin/llvm-config" "-DCMAKE_INSTALL_MESSAGE=LAZY" "-DCMAKE_C_COMPILER=/Users/runner/work/rust/rust/clang+llvm-15.0.7-x86_64-apple-darwin21.0/bin/clang" "-DCMAKE_CXX_COMPILER=/Users/runner/work/rust/rust/clang+llvm-15.0.7-x86_64-apple-darwin21.0/bin/clang++" "-DCMAKE_ASM_COMPILER=/Users/runner/work/rust/rust/clang+llvm-15.0.7-x86_64-apple-darwin21.0/bin/clang" "-DCMAKE_C_FLAGS=-ffunction-sections -fdata-sections -fPIC -m64 --target=x86_64-apple-macosx -fdebug-prefix-map=/Users/runner/work/rust/rust=/rustc/llvm" "-DCMAKE_CXX_FLAGS=-ffunction-sections -fdata-sections -fPIC -m64 --target=x86_64-apple-macosx -stdlib=libc++ -fdebug-prefix-map=/Users/runner/work/rust/rust=/rustc/llvm" "-DCMAKE_SHARED_LINKER_FLAGS= -static-libstdc++" "-DCMAKE_MODULE_LINKER_FLAGS= -static-libstdc++" "-DCMAKE_EXE_LINKER_FLAGS= -static-libstdc++" "-DCMAKE_INSTALL_PREFIX=/Users/runner/work/rust/rust/build/x86_64-apple-darwin/native/sanitizers" "-DCMAKE_ASM_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64 --target=x86_64-apple-macosx -mmacosx-version-min=10.12" "-DCMAKE_BUILD_TYPE=Release"
CMake Deprecation Warning at /Users/runner/work/rust/rust/src/llvm-project/cmake/Modules/CMakePolicy.cmake:6 (cmake_policy):
  The OLD behavior for policy CMP0116 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
---
ld: warning: '/Users/runner/work/rust/rust/build/x86_64-apple-darwin/native/sanitizers/build/lib/lsan/CMakeFiles/clang_rt.lsan_osx_dynamic.dir/lsan_malloc_mac.cpp.o' has malformed LC_DYSYMTAB, expected 54 undefined symbols to start at index 56, found 51 undefined symbols starting at index 56
ld: warning: '/Users/runner/work/rust/rust/build/x86_64-apple-darwin/native/sanitizers/build/lib/lsan/CMakeFiles/clang_rt.lsan_osx_dynamic.dir/lsan_posix.cpp.o' has malformed LC_DYSYMTAB, expected 25 undefined symbols to start at index 16, found 22 undefined symbols starting at index 16
ld: warning: '/Users/runner/work/rust/rust/build/x86_64-apple-darwin/native/sanitizers/build/lib/lsan/CMakeFiles/clang_rt.lsan_osx_dynamic.dir/lsan_thread.cpp.o' has malformed LC_DYSYMTAB, expected 34 undefined symbols to start at index 34, found 31 undefined symbols starting at index 34
cargo:root=/Users/runner/work/rust/rust/build/x86_64-apple-darwin/native/sanitizers
running: cd "/Users/runner/work/rust/rust/build/x86_64-apple-darwin/native/sanitizers/build" && CMAKE_PREFIX_PATH="" DESTDIR="" LC_ALL="C" "cmake" "/Users/runner/work/rust/rust/src/llvm-project/compiler-rt" "-DCMAKE_OSX_ARCHITECTURES=x86_64" "-G" "Ninja" "-DCMAKE_C_COMPILER_TARGET=x86_64-apple-darwin" "-DCOMPILER_RT_BUILD_BUILTINS=OFF" "-DCOMPILER_RT_BUILD_CRT=OFF" "-DCOMPILER_RT_BUILD_LIBFUZZER=OFF" "-DCOMPILER_RT_BUILD_PROFILE=OFF" "-DCOMPILER_RT_BUILD_SANITIZERS=ON" "-DCOMPILER_RT_BUILD_XRAY=OFF" "-DCOMPILER_RT_DEFAULT_TARGET_ONLY=ON" "-DCOMPILER_RT_USE_LIBCXX=OFF" "-DLLVM_CONFIG_PATH=/Users/runner/work/rust/rust/build/x86_64-apple-darwin/llvm/bin/llvm-config" "-DCMAKE_INSTALL_MESSAGE=LAZY" "-DCMAKE_C_COMPILER=/Users/runner/work/rust/rust/clang+llvm-15.0.7-x86_64-apple-darwin21.0/bin/clang" "-DCMAKE_CXX_COMPILER=/Users/runner/work/rust/rust/clang+llvm-15.0.7-x86_64-apple-darwin21.0/bin/clang++" "-DCMAKE_ASM_COMPILER=/Users/runner/work/rust/rust/clang+llvm-15.0.7-x86_64-apple-darwin21.0/bin/clang" "-DCMAKE_C_FLAGS=-ffunction-sections -fdata-sections -fPIC -m64 --target=x86_64-apple-macosx -fdebug-prefix-map=/Users/runner/work/rust/rust=/rustc/llvm" "-DCMAKE_CXX_FLAGS=-ffunction-sections -fdata-sections -fPIC -m64 --target=x86_64-apple-macosx -stdlib=libc++ -fdebug-prefix-map=/Users/runner/work/rust/rust=/rustc/llvm" "-DCMAKE_SHARED_LINKER_FLAGS= -static-libstdc++" "-DCMAKE_MODULE_LINKER_FLAGS= -static-libstdc++" "-DCMAKE_EXE_LINKER_FLAGS= -static-libstdc++" "-DCMAKE_INSTALL_PREFIX=/Users/runner/work/rust/rust/build/x86_64-apple-darwin/native/sanitizers" "-DCMAKE_ASM_FLAGS= -ffunction-sections -fdata-sections -fPIC -m64 --target=x86_64-apple-macosx -mmacosx-version-min=10.12" "-DCMAKE_BUILD_TYPE=Release"
CMake Deprecation Warning at /Users/runner/work/rust/rust/src/llvm-project/cmake/Modules/CMakePolicy.cmake:6 (cmake_policy):
  The OLD behavior for policy CMP0116 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
---
  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: '/Users/runner/work/rust/rust/build/aarch64-apple-ios/native/sanitizers/build/CMakeFiles/CMakeScratch/TryCompile-bW49rT'
    
    Run Build Command(s): /usr/local/bin/ninja -v cmTC_c661d
    [1/2] /Users/runner/work/rust/rust/clang+llvm-15.0.7-x86_64-apple-darwin21.0/bin/clang --target=aarch64-apple-ios   -fPIC --target=arm64-apple-ios -miphoneos-version-min=17.2 -isysroot /Applications/Xcode_15.2.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.2.sdk -fdebug-prefix-map=/Users/runner/work/rust/rust=/rustc/llvm  -arch arm64 -isysroot /Applications/Xcode_15.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.2.sdk -mmacosx-version-min=10.12 -MD -MT CMakeFiles/cmTC_c661d.dir/testCCompiler.c.o -MF CMakeFiles/cmTC_c661d.dir/testCCompiler.c.o.d -o CMakeFiles/cmTC_c661d.dir/testCCompiler.c.o -c /Users/runner/work/rust/rust/build/aarch64-apple-ios/native/sanitizers/build/CMakeFiles/CMakeScratch/TryCompile-bW49rT/testCCompiler.c
    FAILED: CMakeFiles/cmTC_c661d.dir/testCCompiler.c.o 
    /Users/runner/work/rust/rust/clang+llvm-15.0.7-x86_64-apple-darwin21.0/bin/clang --target=aarch64-apple-ios   -fPIC --target=arm64-apple-ios -miphoneos-version-min=17.2 -isysroot /Applications/Xcode_15.2.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.2.sdk -fdebug-prefix-map=/Users/runner/work/rust/rust=/rustc/llvm  -arch arm64 -isysroot /Applications/Xcode_15.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.2.sdk -mmacosx-version-min=10.12 -MD -MT CMakeFiles/cmTC_c661d.dir/testCCompiler.c.o -MF CMakeFiles/cmTC_c661d.dir/testCCompiler.c.o.d -o CMakeFiles/cmTC_c661d.dir/testCCompiler.c.o -c /Users/runner/work/rust/rust/build/aarch64-apple-ios/native/sanitizers/build/CMakeFiles/CMakeScratch/TryCompile-bW49rT/testCCompiler.c
    clang-15: error: invalid argument '-mmacosx-version-min=10.12' not allowed with '-miphoneos-version-min=17.2'
    clang-15: warning: overriding '-mmacosx-version-min=10.12' option with '--target=arm64-apple-ios' [-Woverriding-t-option]
    clang-15: warning: using sysroot for 'MacOSX' but targeting 'iPhone' [-Wincompatible-sysroot]
    ninja: build stopped: subcommand failed.
    
    

  

```

</details>

--- a/src/bootstrap/download-ci-llvm-stamp
+++ b/src/bootstrap/download-ci-llvm-stamp
@@ -1,4 +1,4 @@
 Change this file to make users of the `download-ci-llvm` configuration download
 a new version of LLVM from CI, even if the LLVM submodule hasn’t changed.
 
-Last change is for: https://github.com/rust-lang/rust/pull/134740
+Last change is for: https://github.com/rust-lang/rust/pull/138784

--- a/src/bootstrap/src/core/build_steps/llvm.rs
+++ b/src/bootstrap/src/core/build_steps/llvm.rs
@@ -645,10 +645,17 @@ fn configure_cmake(
     if !builder.is_builder_target(target) {
         cfg.define("CMAKE_CROSSCOMPILING", "True");
 
+        // NOTE: Ideally, we wouldn't have to do this, and `cmake-rs` would just handle it for us.
+        // But it currently determines this based on the `CARGO_CFG_TARGET_OS` environment variable,
+        // which isn't set when compiling outside `build.rs` (like bootstrap is).
+        //
+        // So for now, we define `CMAKE_SYSTEM_NAME` ourselves, to panicking in `cmake-rs`.
         if target.contains("netbsd") {
             cfg.define("CMAKE_SYSTEM_NAME", "NetBSD");
         } else if target.contains("dragonfly") {
             cfg.define("CMAKE_SYSTEM_NAME", "DragonFly");
+        } else if target.contains("openbsd") {
+            cfg.define("CMAKE_SYSTEM_NAME", "OpenBSD");
         } else if target.contains("freebsd") {
             cfg.define("CMAKE_SYSTEM_NAME", "FreeBSD");
         } else if target.is_windows() {
@@ -659,10 +666,27 @@ fn configure_cmake(
             cfg.define("CMAKE_SYSTEM_NAME", "SunOS");
         } else if target.contains("linux") {
             cfg.define("CMAKE_SYSTEM_NAME", "Linux");
+        } else if target.contains("darwin") {
+            // macOS
+            cfg.define("CMAKE_SYSTEM_NAME", "Darwin");
+        } else if target.contains("ios") {
+            cfg.define("CMAKE_SYSTEM_NAME", "iOS");
+        } else if target.contains("tvos") {
+            cfg.define("CMAKE_SYSTEM_NAME", "tvOS");
+        } else if target.contains("visionos") {
+            cfg.define("CMAKE_SYSTEM_NAME", "visionOS");
+        } else if target.contains("watchos") {
+            cfg.define("CMAKE_SYSTEM_NAME", "watchOS");
+        } else if target.contains("none") {
+            // "none" should be the last branch
+            cfg.define("CMAKE_SYSTEM_NAME", "Generic");
         } else {
             builder.info(&format!(
                 "could not determine CMAKE_SYSTEM_NAME from the target `{target}`, build may fail",
             ));
+            // Fallback, set `CMAKE_SYSTEM_NAME` anyhow to avoid the logic `cmake-rs` tries, and
+            // to avoid CMAKE_SYSTEM_NAME being inferred from the host.
+            cfg.define("CMAKE_SYSTEM_NAME", "Generic");
         }
 
         // When cross-compiling we should also set CMAKE_SYSTEM_VERSION, but in

--- a/src/bootstrap/src/core/build_steps/llvm.rs
+++ b/src/bootstrap/src/core/build_steps/llvm.rs
@@ -385,9 +385,6 @@ impl Step for Llvm {
             || target.contains("apple-watchos")
             || target.contains("apple-visionos")
         {
-            // These two defines prevent CMake from automatically trying to add a MacOSX sysroot, which leads to a compiler error.
-            cfg.define("CMAKE_OSX_SYSROOT", "/");
-            cfg.define("CMAKE_OSX_DEPLOYMENT_TARGET", "");
             // Prevent cmake from adding -bundle to CFLAGS automatically, which leads to a compiler error because "-bitcode_bundle" also gets added.
             cfg.define("LLVM_ENABLE_PLUGINS", "OFF");
             // Zlib fails to link properly, leading to a compiler error.
@@ -703,6 +700,10 @@ fn configure_cmake(
                 //
                 // So for now we set it to "Darwin" on all Apple platforms.
                 cfg.define("CMAKE_SYSTEM_NAME", "Darwin");
+
+                // These two defines prevent CMake from automatically trying to add a MacOSX sysroot, which leads to a compiler error.
+                cfg.define("CMAKE_OSX_SYSROOT", "/");
+                cfg.define("CMAKE_OSX_DEPLOYMENT_TARGET", "");
             }
 
             // Make sure that CMake does not build universal binaries on macOS.

